lib/Transforms/YosysOptimizer/YosysOptimizer.cpp

```diff
@@ -229,6 +229,44 @@ LogicalResult convertOpResults(secret::GenericOp op,
   return success();
 }
 
+void constantPassThrough(secret::GenericOp genericOp,
```

Comment:
Is there a reason not to implement this as a pattern? I suspect it would be useful in other contexts as well, and it would remove the need to walk.

Reply:
You will still need to walk because anchoring it on a GenericOp doesn't allow me to see outside the context of the generic (or so it seemed to not work when I did that).

But in general, the reason why I didn't make it a pattern was because patterns are applied and automatically the IR si folded by the pattern rewriter infra. That nullifies this pattern because then these constants are folded and moved outside the generic body because generics aren't isolated from above.

Reply:
I am happy to move it to a helper in SecretPatterns though!

Reply:
ah, ok, so what I'd say we should do is make it a helper function in SecretPatterns or similar, and expose it as its own pass that does nothing but call the helper, if that's not too much to ask.

Reply:
done!